cbcontainers/state/hardening/adapters/validating_webhook.go

```diff
@@ -0,0 +1,352 @@
+package adapters
+
+import (
+	admissionsV1 "k8s.io/api/admissionregistration/v1"
+	admissionsV1Beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ValidatingWebhookConfigurationAdapter interface {
+	// SetLabels will update the validating webhook configuration's labels
+	SetLabels(labels map[string]string)
+	// GetWebhooks will return the webhooks attached to this configuration instance wrapped in an adapter.
+	// The returned adapters wrap pointers to the original values so any modifications are propagated on the objects
+	// For adding/removing webhooks to the list, use SetWebhooks
+	GetWebhooks() []ValidatingWebhookAdapter
+	// SetWebhooks will replace the configuration's webhooks with the provided list
+	// Any webhooks that don't match the configuration adapter's API version are ignored
+	// E.g. passing a list of {v1, v1beta1, v1} wrapped webhooks to a v1 ValidatingWebhookAdapter will only set the 2 v1 webhooks and ignore the v1beta1
+	SetWebhooks([]ValidatingWebhookAdapter)
+}
+
+type ValidatingWebhookAdapter interface {
+	GetName() string
```

Comment:
using getters and setters is not discouraged in Go, but the naming convention is to avoid the `Get` part of the method name - https://golang.org/doc/effective_go#Getters